handlers/services.go

```diff
@@ -102,8 +103,11 @@ func ServiceDetails(w http.ResponseWriter, r *http.Request) {
 		go func(istioConfigValidations *models.IstioValidations, err *error) {
 			defer wg.Done()
 			istioConfigValidationResults, errValidations := business.Validations.GetValidations(namespace, service)
-			if errValidations != nil && *err == nil {
-				*err = errValidations
+			if errValidations != nil {
+				if *err == nil {
```

Comment:
BTW that part doesn't look thread-safe

Reply:
It isn't, but the end result will still always be correct (as it doesn't matter which thread wins - one will).

Reply:
The error handling is quite confusing here. `err` is read / written in goroutine as well as in the main thread for different purposes. I guess we should clear that up.

Reply:
To clarify, if the error was created by the main thread, it will exit at line 118 but if the validation fails it will most likely exit at line 128 (as there's a wg.Wait()). 

In any case, both will respond with the error so why log it?

Reply:
I don't think it matters to the caller, right? They will get an error that prevented the call in any case, it doesn't matter if both failed.

Reply:
Hmm in my understanding, the validation failure is sent as response only if the call to `business.Validations.GetValidations(namespace, service)` returns after `business.Svc.GetService(namespace, service, rateInterval, queryTime, requestToken)` ... which is not guaranteed

Reply:
It's guaranteed in line 124 by the ``wg.Wait()`` if the validations were requested. If error happened with the ``GetService`` then we don't wait for the validations as it doesn't make any sense.

Reply:
Ok, I'll take another look at it with the fresh eyes of a Monday morning :-)

Reply:
So I still don't get it. Here's the scenario that I think results in ignoring the error:

1. `GetValidations` call begins (l.105)
2. `GetServices` call begins (l.122)
3. `GetValidations` call ends, `err` filled with a validation error (l.108)
4. `GetServices` call ends, `err` being overwritten with `nil`, given there's no error while getting the service (l.122)
5. `wg.Wait` is not blocking as the goroutine has already finished (l.124)
6. `err` is `nil` => responding success & losing track of the validation error that happened.

Makes sense?

Reply:
Ah yeah, in that scenario it does, GetService can overwrite the GetValidations' error.